src/core/queries.ml

```diff
@@ -50,45 +54,55 @@ let handle_query : Sig_state.t -> proof_state option -> p_query -> unit =
           | Some [] ->
               if Eval.eq_modulo ctxt t u = must_fail then
                 fatal q.pos "Assertion failed."
-      end
+      end;
+      None
   | P_query_debug(e,s) ->
       (* Just update the option, state not modified. *)
       Console.set_debug e s;
-      out 3 "(flag) debug → %s%s\n" (if e then "+" else "-") s
+      out 3 "(flag) debug → %s%s\n" (if e then "+" else "-") s;
+      None
   | P_query_verbose(i) ->
       (* Just update the option, state not modified. *)
       if Timed.(!Console.verbose) = 0 then
         (Timed.(Console.verbose := i); out 1 "(flag) verbose → %i\n" i)
       else
-        (out 1 "(flag) verbose → %i\n" i; Timed.(Console.verbose := i))
+        (out 1 "(flag) verbose → %i\n" i; Timed.(Console.verbose := i));
+      None
   | P_query_flag(id,b) ->
       (* We set the value of the flag, if it exists. *)
       (try Console.set_flag id b
        with Not_found -> wrn q.pos "Unknown flag \"%s\"." id);
-      out 3 "(flag) %s → %b\n" id b
+      out 3 "(flag) %s → %b\n" id b;
+      None
   | P_query_infer(pt, cfg) ->
       let env = Proof.focus_env ps in
       let ctxt = Env.to_ctxt env in
       let scope = Scope.scope_term Public ss env in
       let t = scope pt in
       let a = Infer.infer Unif.solve_noexn pt.pos ctxt t in
-      out 1 "(infr) %a : %a\n" pp_term t pp_term (Eval.eval cfg ctxt a)
+      let res = Eval.eval cfg ctxt a in
+      out 1 "(infr) %a : %a\n" pp_term t pp_term res;
+      Some (Format.asprintf "%a : %a" pp_term t pp_term res)
   | P_query_normalize(pt, cfg) ->
       let env = Proof.focus_env ps in
       let ctxt = Env.to_ctxt env in
       let scope = Scope.scope_term Public ss env in
       let t = scope pt in
       (* Check that [t] is typable. *)
       ignore (Infer.infer Unif.solve_noexn pt.pos ctxt t);
-      out 1 "(comp) %a\n" pp_term (Eval.eval cfg ctxt t)
+      out 1 "(comp) %a\n" pp_term (Eval.eval cfg ctxt t);
+      Some (Format.asprintf "%a" pp_term (Eval.eval cfg ctxt t))
   | P_query_prover(s) ->
-      Timed.(Why3_tactic.default_prover := s)
+      Timed.(Why3_tactic.default_prover := s);
+      None
   | P_query_prover_timeout(n) ->
-      Timed.(Why3_tactic.timeout := n)
+      Timed.(Why3_tactic.timeout := n);
+      None
   | P_query_print(None) ->
       (match ps with
-       | None -> fatal q.pos "Not in a proof."
-       | Some ps -> out 1 "%a" Proof.pp_goals ps)
+       | None -> None
```

Comment:
Sorry but, finally, I think that we should keep `| None -> fatal q.pos "Not in a proof."` as this is incorrect to use `print` with no argument outside a proof.